### PR TITLE
30 api to add poly flow builder flowchart templates to the db

### PR DESCRIPTION
**Pull request by ARES-79**

### CODE REVIEW

**Comment by JennyFerriol** [comment]
can you please add a unit test for this

File: src/frontend/src/JSON/TermData.ts [hunk 1]
```diff
@@ -120,4 +120,13 @@ export const exampleTermData = [
     termName: 'Spring 2030',
     classes: []
   }
-];
\ No newline at end of file
+];
+
+// POST http://localhost:8080/FlowchartTemplates
```
Comment (archie1802) [inline src/frontend/src/JSON/TermData.ts:125]: Can you leave a comment explaining what this is and also in the future we should figure out a cleaner way to try and do this so were not writing json lines by hand

File: src/main/java/com/Q2S/Q2S_Senior_Project/Controllers/FlowchartTemplateController.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+@RestController
+public class FlowchartTemplateController {
+
+    private final FlowchartTemplateRepo flowchartTemplateRepo;
+
+    FlowchartTemplateController(FlowchartTemplateRepo flowchartTemplateRepo){
+        this.flowchartTemplateRepo = flowchartTemplateRepo;
+    }
+    @CrossOrigin(origins = "http://localhost:3000")
+    @PostMapping("/FlowchartTemplates/fromScratch")
```
Comment (archie1802) [inline src/main/java/com/Q2S/Q2S_Senior_Project/Controllers/FlowchartTemplateController.java:27]: Pretty sure api's should have a url "/api/nameofapi" so change format to that

File: src/main/java/com/Q2S/Q2S_Senior_Project/Controllers/FlowchartTemplateController.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        FlowchartTemplate template = new FlowchartTemplate();
+//        template.setId(data.getId());
+        template.setCatalog(data.getCatalog());
+        template.setMajor(data.getMajorName());
+        template.setConcentration(data.getConcName());
+        template.setFlowchart(content);
+        return template;
+    }
+
+    @PostMapping("/FlowchartTemplates")
```
Comment (archie1802) [inline src/main/java/com/Q2S/Q2S_Senior_Project/Controllers/FlowchartTemplateController.java:79]: Same for all these api mappings

**Comment by ARES-79** [comment]
> can you please add a unit test for this

Yes, I will refactor the getFlowchartTemplate() function to be more easily testable.

**Review by archie1802 — APPROVED** [review]

**Pull request merged**